### DISCUSSION TITLE
Help Nordvpn config outdated/wrong

### DISCUSSION THREAD

**Discussion by WaaromZoMoeilijk** [qdm12/gluetun#702] [Q&A]
Hello,

Thanks for this great project.

It seems there are a lot of unresolvable (DNS) hostnames in the `servers.json`to which i can't connect anymore.
I cannot just edit the file with the one server I need since startup creates a fresh `servers.json`
_Note that the connection works on Gluetun (Regular WAN without VPN), I try to resolve the hostnames on my desktop via CLI_

Using Nordvpn. 

Are there some parameters or filters for which servers get pulled? I'd really like to pick a speciality server from NordVPN to connect to, eg onion. But the only one offered in my country in `servers.json` is nl-onion5.nordvpn.com which doesn't exist anymore.

Please advise.

**Comment by qdm12** [comment]
See https://github.com/qdm12/gluetun/wiki/Updating-servers 😉 

All the server information gets stored in servers.json.

If you want you can filter out servers at runtime, see https://github.com/qdm12/gluetun/wiki/NordVPN#optional-environment-variables

**Comment by iluvatyr** [comment]
Where did you see that there are unresolvable DNS hostnames in the servers.json Waarom? Would like to know how to see that

**Reply by qdm12** [reply]
He probably tried to resolve some with `nslookup domain.com` I guess? Why are you curious on that? Note they have a ton of hostnames so obviously some will die off with time etc.

**Reply by iluvatyr** [reply]
Because I thought maybe I can check if this is also the cause for my problem.

**Reply by WaaromZoMoeilijk** [reply]
Well my server of choice would be nl-onion6.nordvpn.com, this one is in the version on github. 
But whenever I grep nl-onion6.nordvpn.com in my servers ,server.json then there's no entry.

As for the hostnames stated in my servers version of server.json, the ones I actually want to use, do not resolve since they are outdated, I presume. I've taken the hostname from the server.json file and  did a `host` or `nslookup` on my desktop on them.

Nordvpn only maintains a current list on their API so its weird to see these entries there, if the list auto updates... right?

Filtering with hostname nl-onion6.nordvpn.com doesn't work either.

It would be awesome if we could just fill in our custom server for use in server.json

ps running with tag latest on docker.

**Comment by WaaromZoMoeilijk** [comment]
End goal: use NL based onion server

TL;DR  server `nl-onion6.nordvpn.com` not available  in `docker containers server.json` while it is listed in `githubs servers.json`

Long version:
See when I use `nl-onion6.nordvpn.com` as SERVER_HOSTNAME it doesn't match with any of the available servers generated: https://hastebin.waaromzomoeilijk.nl/udavugajun.pl
Scroll to the right to see the list of available servers

```
ERROR cannot read VPN settings: cannot read provider settings: environment variable SERVER_HOSTNAME: at least one value is not within the accepted values: invalid values found: value "nl-onion6.nordvpn.com" at position 1; possible values are: LINK ABOVE
```
Only `nl-onion5.nordvpn.com` is available in the servers.json as shown in the command output in the link above.
But when trying to resolve `nl-onion5.nordvpn.com` its not in use anymore:
```
root@pve# host nl-onion5.nordvpn.com
root@pve# nslookup nl-onion5.nordvpn.com
Non-authoritative answer:
*** Can't find nl-onion5.nordvpn.com: No answer
```

But when I try `curl -s https://raw.githubusercontent.com/qdm12/gluetun/master/internal/storage/servers.json | grep nl-onion6.nordvpn.com` 
I do get the entry back with `grep`:
```
curl -s https://raw.githubusercontent.com/qdm12/gluetun/master/internal/storage/servers.json | grep nl-onion6.nordvpn.com
        "hostname": "nl-onion6.nordvpn.com",
```

It seems to not auto update the Nordvpn servers correctly since these are the actual currently available servers:
```
root@desktop:~# curl --silent "https://api.nordvpn.com/v1/servers"  | jq --raw-output '.[] | select(.locations[].country.name == "Netherlands") | .hostname' | grep 'nl-onion'
nl-onion4.nordvpn.com
nl-onion6.nordvpn.com
```
Where nl-onion5 is not on the list anymore

This is my docker-compose config. Networking and DNS resolving works fine in the container and there are servers that do work, just don't fit my needs.
```
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    cap_add:
      - NET_ADMIN
    volumes:
      - /mnt/SSD/Docker/VPN:/gluetun
    environment:
      # See https://github.com/qdm12/gluetun/wiki
      - VPNSP=nordvpn
      - OPENVPN_USER=USER
      - OPENVPN_PASSWORD=PASS
      - REGION=Netherlands # Tried on and off
      - SERVER_HOSTNAME=nl-onion6.nordvpn.com
      - UPDATE_PERIOD=24h
      - OPENVPN_VERBOSITY=6
      - FIREWALL_OUTBOUND_SUBNETS=x.x.x.x/24
      - FIREWALL=off
      - TZ=Europe/Amsterdam
      - VLAN=X
      - DOT=off
      - DNS_PLAINTEXT_ADDRESS=x.x.x.x
    networks:
        vlanX:
            ipv4_address: x.x.x.x
    dns:
      - "x.x.x.x"
    dns_search: ""

networks:
  vlanX:
    name: vlanX
    driver: macvlan
    driver_opts:
      parent: vmbrX
    ipam:
      config:
        - subnet: x.x.x.x/24
          gateway: x.x.x.x

```
I've also ran the server update command but still it only reports back `nl-onion5`:
```
root@pve:/mnt/SSD/Docker/VPN# docker run -it --rm -v $pwd:/gluetun qmcgaw/gluetun update -enduser -nordvpn
2021/11/03 10:28:57 INFO creating /gluetun/servers.json with 11225 hardcoded servers
2021/11/03 10:28:57 INFO updating NordVPN servers...
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #388
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #440
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #837
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #752
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #6249
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #6250
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #6251
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #6252
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #6253
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #6254
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands SOCKS #1
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands SOCKS #2
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands SOCKS #3
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands SOCKS #4
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Arab Emirates #49
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Switzerland #316
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1010
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: France #726
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Italy #208
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Japan #596
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #904
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Singapore #509
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Kingdom #2182
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8023
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8024
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8025
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8026
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8027
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #1
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #2
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #3
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #4
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #5
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Arab Emirates #50
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Switzerland #317
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1012
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Italy #209
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Japan #597
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Kingdom #2188
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Kingdom #2189
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8088
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8089
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8090
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8091
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8092
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8093
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8094
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Arab Emirates #51
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1013
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1014
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Japan #600
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #909
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Sweden #540
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Kingdom #2190
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Kingdom #2191
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8135
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8136
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8138
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8139
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8140
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8141
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #6
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #7
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #8
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #9
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #11
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #12
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #13
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #14
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States SOCKS #15
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Sweden SOCKS #1
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Sweden SOCKS #2
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Sweden SOCKS #3
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Sweden SOCKS #4
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Sweden SOCKS #5
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Ireland SOCKS #1
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Ireland SOCKS #2
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Ireland SOCKS #3
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Ireland SOCKS #4
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Ireland SOCKS #5
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Arab Emirates #52
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Japan #601
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Sweden #541
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Kingdom #2192
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United Arab Emirates #53
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: France #782
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: France #783
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: France #784
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: France #785
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: France #786
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #947
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #948
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands #950
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Switzerland #318
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Turkey #49
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Turkey #50
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Canada #1530
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Canada #1531
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Canada #1532
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Canada #1533
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8800
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #8801
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9286
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9287
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9288
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9289
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9290
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9292
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9293
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9294
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Hong Kong #281
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Hong Kong #282
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1048
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1049
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1050
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Germany #1051
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands SOCKS #7
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: Netherlands SOCKS #8
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9466
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9467
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9468
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9469
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9470
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9471
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9472
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9473
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9474
2021/11/03 10:29:01 WARN NordVPN: server does not support TCP and UDP for openvpn: United States #9475
root@pve:/mnt/SSD/Docker/VPN# cat servers.json | grep 'nl-onion'
        "hostname": "nl-onion5.nordvpn.com",

```

Also the same goes for `nl-se13.nordvpn.com` and a few other which i've forgot about :sweat: 
```
root@pve:/mnt/SSD/Docker/VPN# cat servers.json | grep 'nl-se13'
root@pve:/mnt/SSD/Docker/VPN# 
```

```
root@pve:/mnt/SSD/Docker/VPN# curl -s https://raw.githubusercontent.com/qdm12/gluetun/master/internal/storage/servers.json | grep 'nl-se13'
        "hostname": "nl-se13.nordvpn.com",
```

**Reply by qdm12** [reply]
Hey sorry for 6 months delay, can you try again on the latest image? I still see those servers and they should be working. Let me know and I'll create an issue if this is still not working.
